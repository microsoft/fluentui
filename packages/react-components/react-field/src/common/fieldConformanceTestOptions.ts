/**
 * Defaults for the isConformant tests of field components.
 *
 * @internal
 */
export const fieldConformanceTestOptions = {
  'has-static-classnames': [
    {
      props: {
        label: 'label text',
        validationState: 'error',
        validationMessage: 'validation message text',
        hint: 'hint text',
      },
    },
  ],
};
