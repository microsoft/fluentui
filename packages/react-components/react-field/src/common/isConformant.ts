import { isConformant as baseIsConformant } from '@fluentui/react-conformance';
import type { IsConformantOptions, TestObject } from '@fluentui/react-conformance';
import griffelTests from '@fluentui/react-conformance-griffel';

export function isConformant<TProps = {}>(
  testInfo: Omit<IsConformantOptions<TProps>, 'componentPath'> & { componentPath?: string },
) {
  const defaultOptions: Partial<IsConformantOptions<TProps>> = {
    componentPath: module!.parent!.filename.replace('.test', ''),
    extraTests: griffelTests as TestObject<TProps>,

    // Field-specific defaults
    primarySlot: 'control' as keyof TProps,
    testOptions: testInfo.testOptions?.['has-static-classnames']
      ? undefined
      : {
          'has-static-classnames': [
            {
              props: {
                label: 'label text',
                validationState: 'error',
                validationMessage: 'validation message text',
                hint: 'hint text',
              },
            },
          ],
        },
  };

  baseIsConformant(defaultOptions, testInfo);
}
