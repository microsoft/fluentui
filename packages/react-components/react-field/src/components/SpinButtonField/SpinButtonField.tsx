import * as React from 'react';
import { SpinButton } from '@fluentui/react-spinbutton';
import type { ForwardRefComponent } from '@fluentui/react-utilities';
import type { FieldProps } from '../../Field';
import { getFieldClassNames, renderField_unstable, useFieldStyles_unstable, useField_unstable } from '../../Field';

export type SpinButtonFieldProps = FieldProps<typeof SpinButton>;

export const spinButtonFieldClassNames = getFieldClassNames('SpinButtonField');

export const SpinButtonField: ForwardRefComponent<SpinButtonFieldProps> = React.forwardRef((props, ref) => {
  const state = useField_unstable(props, ref, { component: SpinButton, classNames: spinButtonFieldClassNames });
  useFieldStyles_unstable(state);
  return renderField_unstable(state);
});

SpinButtonField.displayName = 'SpinButtonField';
