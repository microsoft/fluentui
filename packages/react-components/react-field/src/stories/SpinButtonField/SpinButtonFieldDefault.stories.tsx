import * as React from 'react';
import type { SpinButtonFieldProps } from '@fluentui/react-field';
import { SpinButtonField } from '@fluentui/react-field';

export const Default = (props: Partial<SpinButtonFieldProps>) => (
  <SpinButtonField
    label="Example spin button field"
    validationState="success"
    validationMessage="This is a success message"
    hint="This is a hint message"
    {...props}
  />
);
