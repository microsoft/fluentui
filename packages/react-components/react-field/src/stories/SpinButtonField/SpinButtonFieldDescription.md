SpinButtonField adds a Label, validation message, and hint text to the SpinButton control.
