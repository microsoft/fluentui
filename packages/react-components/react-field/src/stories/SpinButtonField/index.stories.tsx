import { SpinButtonField } from '@fluentui/react-field';

import descriptionMd from './SpinButtonFieldDescription.md';

export { Default } from './SpinButtonFieldDefault.stories';

export default {
  title: 'Preview Components/Field/SpinButtonField',
  component: SpinButtonField,
  parameters: {
    docs: {
      description: {
        component: descriptionMd,
      },
    },
  },
};
