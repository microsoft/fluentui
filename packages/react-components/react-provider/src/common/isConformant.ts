import { isConformant as baseIsConformant } from '@fluentui/react-conformance';
import type { IsConformantOptions } from '@fluentui/react-conformance';

export function isConformant<TProps = {}>(
  testInfo: Omit<IsConformantOptions<TProps>, 'componentPath'> & { componentPath?: string },
) {
  const defaultOptions: Partial<IsConformantOptions<TProps>> = {
    componentPath: require.main?.filename.replace('.test', ''),
  };

  baseIsConformant(defaultOptions, testInfo);
}
