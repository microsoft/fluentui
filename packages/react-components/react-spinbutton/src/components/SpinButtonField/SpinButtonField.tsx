/* eslint-disable deprecation/deprecation */
import { DeprecatedFieldProps, getDeprecatedFieldClassNames, makeDeprecatedField } from '@fluentui/react-field';
import type { ForwardRefComponent } from '@fluentui/react-utilities';
import { SpinButton, spinButtonClassNames, SpinButtonProps } from '../../SpinButton';

/** @deprecated Use Field with SpinButton: `<Field><SpinButton /></Field>` */
export type SpinButtonFieldProps = DeprecatedFieldProps<SpinButtonProps>;
/** @deprecated Use Field with SpinButton: `<Field><SpinButton /></Field>` */
export const spinButtonFieldClassNames = getDeprecatedFieldClassNames(spinButtonClassNames.root);
/** @deprecated Use Field with SpinButton: `<Field><SpinButton /></Field>` */
export const SpinButtonField: ForwardRefComponent<SpinButtonFieldProps> = makeDeprecatedField(SpinButton);
