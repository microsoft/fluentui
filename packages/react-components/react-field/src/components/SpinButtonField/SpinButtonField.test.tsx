import { isConformant } from '../../common/isConformant';
import { SpinButtonField } from './SpinButtonField';

describe('SpinButtonField', () => {
  isConformant({
    Component: SpinButtonField,
    displayName: 'SpinButtonField',
  });

  // Most functionality is tested by Field.test.tsx, and SpinButton's tests
});
