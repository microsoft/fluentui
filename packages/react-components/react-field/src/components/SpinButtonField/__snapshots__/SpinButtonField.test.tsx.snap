// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpinButtonField renders a default state 1`] = `
<div>
  <div
    class="fui-SpinButtonField"
  >
    <span
      class="fui-Input fui-SpinButton fui-SpinButtonField__control"
    >
      <input
        appearance="outline"
        aria-valuenow="0"
        autocomplete="off"
        class="fui-SpinButton__input fui-Input__input"
        role="spinbutton"
        type="text"
        value="0"
      />
      <button
        aria-label="Increment value"
        class="fui-SpinButton__incrementButton"
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class=""
          fill="currentColor"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.15 10.35c.2.2.5.2.7 0L8 6.21l4.15 4.14a.5.5 0 00.7-.7l-4.5-4.5a.5.5 0 00-.7 0l-4.5 4.5a.5.5 0 000 .7z"
            fill="currentColor"
          />
        </svg>
      </button>
      <button
        aria-label="Decrement value"
        class="fui-SpinButton__decrementButton"
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class=""
          fill="currentColor"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.15 5.65c.2-.2.5-.2.7 0L8 9.79l4.15-4.14a.5.5 0 01.7.7l-4.5 4.5a.5.5 0 01-.7 0l-4.5-4.5a.5.5 0 010-.7z"
            fill="currentColor"
          />
        </svg>
      </button>
    </span>
  </div>
</div>
`;
